Missing Xerces include

**Environment:**
- Windows 10
- Visual Studio 2017, 2019
- PDAL commit f4a651c

**Bug**

In the script plugins/e57/libE57Format/CMakeLists.txt the library is linked with Xerces
```
target_link_libraries(E57Format
    PUBLIC
        ${XercesC_LIBRARIES}
)
```
but the include directory is missing. I have appended
```
target_include_directories(E57Format PUBLIC ${XercesC_INCLUDE_DIR})
```
to the end of the script.

```
D:\dev\workspace\test3>SET WORKSPACE=%CD%
D:\dev\workspace\test3>robocopy ..\test2\deps deps /MIR
[...]
D:\dev\workspace\test3>git clone https://github.com/PDAL/PDAL.git src
[...]
D:\dev\workspace\test3>PowerShell -Command "(Get-Content -Path '%WORKSPACE%\src\vendor\arbiter\CMakeLists.txt') -replace 'find_library\(SHLWAPI Shlwapi.lib\)','set(SHLWAPI Shlwapi.lib)' | Set-Content -Path '%WORKSPACE%\src\vendor\arbiter\CMakeLists.txt'"
D:\dev\workspace\test3>mkdir build
D:\dev\workspace\test3>cd build
D:\dev\workspace\test3>cmake -G "Visual Studio 15" -A x64 -D CMAKE_DEBUG_POSTFIX:STRING=d -D CMAKE_INSTALL_PREFIX:PATH=%WORKSPACE%\install -D BUILD_PLUGIN_HDF:BOOL=ON -D BUILD_PLUGIN_E57:BOOL=ON -D BUILD_PLUGIN_I3S:BOOL=ON -D BUILD_PLUGIN_ICEBRIDGE:BOOL=ON -D BUILD_PLUGIN_MATLAB:BOOL=ON -D BUILD_PLUGIN_MBIO:BOOL=ON -D BUILD_PLUGIN_MRSID:BOOL=ON -D WITH_LZMA:BOOL=ON -D WITH_STATIC_LASZIP:BOOL=ON -D WITH_TESTS:BOOL=OFF -D GDAL_INCLUDE_DIR:PATH=%WORKSPACE%\deps\GDAL\install\include -D GDAL_LIBRARY:FILEPATH=debug;%WORKSPACE%\deps\GDAL\install\lib\gdal_i_d.lib;optimized;%WORKSPACE%\deps\GDAL\install\lib\gdal_i.lib -D GEOTIFF_INCLUDE_DIR:PATH=%WORKSPACE%\deps\libgeotiff\install\include -D GEOTIFF_LIBRARY:FILEPATH=debug;%WORKSPACE%\deps\libgeotiff\install\lib\geotiff_d.lib;optimized;%WORKSPACE%\deps\libgeotiff\install\lib\geotiff.lib -D ZLIB_INCLUDE_DIR:PATH=%WORKSPACE%\deps\zlib\install\include -D ZLIB_LIBRARY_DEBUG:FILEPATH=%WORKSPACE%\deps\zlib\install\lib\zlibstaticd.lib -D ZLIB_LIBRARY_RELEASE:FILEPATH=%WORKSPACE%\deps\zlib\install\lib\zlibstatic.lib -D CURL_INCLUDE_DIR:PATH=%WORKSPACE%\deps\curl\install\include -D CURL_LIBRARY:FILEPATH=debug;%WORKSPACE%\deps\curl\install\lib\libcurld_imp.lib;optimized;%WORKSPACE%\deps\curl\install\lib\libcurl_imp.lib -D LASZIP_INCLUDE_DIR:PATH=%WORKSPACE%\deps\LASzip\install\include -D LASZIP_LIBRARY:FILEPATH=debug;%WORKSPACE%\deps\LASzip\install\lib\laszip3staticd.lib;optimized;%WORKSPACE%\deps\LASzip\install\lib\laszip3static.lib -D LIBLZMA_INCLUDE_DIR:PATH=%WORKSPACE%\deps\liblzma\install\include -D LIBLZMA_LIBRARY_DEBUG:FILEPATH=%WORKSPACE%\deps\liblzma\install\lib\liblzmad.lib -D LIBLZMA_LIBRARY_RELEASE:FILEPATH=%WORKSPACE%\deps\liblzma\install\lib\liblzma.lib -D OPENSSL_ROOT_DIR:PATH=%WORKSPACE%\deps\OpenSSL\install -D OPENSSL_INCLUDE_DIR:PATH=%WORKSPACE%\deps\OpenSSL\install\include -D LIBXML2_INCLUDE_DIR:PATH=%WORKSPACE%\deps\libxml2\install\include\libxml2 -D LIBXML2_LIBRARY:FILEPATH=debug;%WORKSPACE%\deps\libxml2\install\lib\Debug\libxml2.lib;optimized;%WORKSPACE%\deps\libxml2\install\lib\Release\libxml2.lib -D LIBXML2_XMLLINT_EXECUTABLE:FILEPATH=%WORKSPACE%\deps\libxml2\install\bin\Release\xmllint.exe -D HDF5_DIR:PATH=%WORKSPACE%\deps\hdf5\install\cmake\hdf5 -D HDF5_C_INCLUDE_DIR:PATH=%WORKSPACE%\deps\hdf5\install\include -D HDF5_DIFF_EXECUTABLE:FILEPATH=%WORKSPACE%\deps\hdf5\install\bin\h5diff.exe -D HDF5_hdf5_LIBRARY_DEBUG:FILEPATH=%WORKSPACE%\deps\hdf5\install\lib\libhdf5_D.lib -D HDF5_hdf5_LIBRARY_RELEASE:FILEPATH=%WORKSPACE%\deps\hdf5\install\lib\libhdf5.lib -D XercesC_INCLUDE_DIR:PATH=%WORKSPACE%\deps\Xerces-C++\install\include -D XercesC_LIBRARY_DEBUG:FILEPATH=%WORKSPACE%\deps\Xerces-C++\install\lib\xerces-c_3D.lib -D XercesC_LIBRARY_RELEASE:FILEPATH=%WORKSPACE%\deps\Xerces-C++\install\lib\xerces-c_3.lib %WORKSPACE%\src
[...]
D:\dev\workspace\test3>MSBuild.exe %WORKSPACE%\build\INSTALL.vcxproj /maxcpucount /nodeReuse:false /nologo /p:Configuration=Debug
[...]
     5>D:\dev\workspace\test3\src\plugins\e57\libE57Format\src\E57XmlParser.cpp(27): fatal error C1083: Datei (Include) kann nicht geöffnet werden: "xercesc/sax2/Attributes.hpp": No such file or directory [D:\dev\workspace\test3\build\plugins\e57\libE57Format\E57Format.vcxproj]
     5>D:\dev\workspace\test3\src\plugins\e57\libE57Format\src\CheckedFile.cpp(557): warning C4996: 'unlink': The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _unlink. See online help for details. [D:\dev\workspace\test3\build\plugins\e57\libE57Format\E57Format.vcxproj]
         C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucrt\corecrt_io.h(553): note: Siehe Deklaration von "unlink"
     5>d:\dev\workspace\test3\src\plugins\e57\libe57format\src\E57XmlParser.h(32): fatal error C1083: Datei (Include) kann nicht geöffnet werden: "xercesc/sax2/DefaultHandler.hpp": No such file or directory (Quelldatei wird kompiliert D:\dev\workspace\test3\src\plugins\e57\libE57Format\src\ImageFileImpl.cpp) [D:\dev\workspace\test3\build\plugins\e57\libE57Format\E57Format.vcxproj]
[...]
       "D:\dev\workspace\test3\build\INSTALL.vcxproj" (Standardziel) (1) ->
       "D:\dev\workspace\test3\build\ALL_BUILD.vcxproj" (Standardziel) (3) ->
       "D:\dev\workspace\test3\build\plugins\e57\libE57Format\E57Format.vcxproj" (Standardziel) (5) ->
       (ClCompile Ziel) ->
         D:\dev\workspace\test3\src\plugins\e57\libE57Format\src\E57XmlParser.cpp(27): fatal error C1083: Datei (Include) kann nicht geöffnet werden: "xercesc/sax2/Attributes.hpp": No such file or directory [D:\dev\workspace\test3\build\plugins\e57\libE57Format\E57Format.vcxproj]
         d:\dev\workspace\test3\src\plugins\e57\libe57format\src\E57XmlParser.h(32): fatal error C1083: Datei (Include) kann nicht geöffnet werden: "xercesc/sax2/DefaultHandler.hpp": No such file or directory (Quelldatei wird kompiliert D:\dev\workspace\test3\src\plugins\e57\libE57Format\src\ImageFileImpl.cpp) [D:\dev\workspace\test3\build\plugins\e57\libE57Format\E57Format.vcxproj]


       "D:\dev\workspace\test3\build\INSTALL.vcxproj" (Standardziel) (1) ->
       "D:\dev\workspace\test3\build\ALL_BUILD.vcxproj" (Standardziel) (3) ->
       "D:\dev\workspace\test3\build\plugins\matlab\libpdal_plugin_filter_matlab.vcxproj" (Standardziel) (9) ->
         d:\dev\workspace\test3\src\plugins\matlab\filters\MatlabFilter.hpp(42): fatal error C1083: Datei (Include) kann nicht geöffnet werden: "json/json.h": No such file or directory (Quelldatei wird kompiliert D:\dev\workspace\test3\src\plugins\matlab\filters\MatlabFilter.cpp) [D:\dev\workspace\test3\build\plugins\matlab\libpdal_plugin_filter_matlab.vcxproj]

    114 Warnung(en)
    3 Fehler

Verstrichene Zeit 00:01:16.09
```
The problem does not disappear when you close the ticket. Maybe you have the Xerces include directory somewhere in your global system variables?

>     5>D:\dev\workspace\test3\src\plugins\e57\libE57Format\src\E57XmlParser.cpp(27): fatal error C1083: Datei (Include) kann nicht geöffnet werden: "xercesc/sax2/Attributes.hpp": No such file or directory [D:\dev\workspace\test3\build\plugins\e57\libE57Format\E57Format.vcxproj]
>     5>D:\dev\workspace\test3\src\plugins\e57\libE57Format\src\CheckedFile.cpp(557): warning C4996: 'unlink': The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _unlink. See online help for details. [D:\dev\workspace\test3\build\plugins\e57\libE57Format\E57Format.vcxproj]
>         C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucrt\corecrt_io.h(553): note: Siehe Deklaration von "unlink"
>     5>d:\dev\workspace\test3\src\plugins\e57\libe57format\src\E57XmlParser.h(32): fatal error C1083: Datei (Include) kann nicht geöffnet werden: "xercesc/sax2/DefaultHandler.hpp": No such file or directory (Quelldatei wird kompiliert D:\dev\workspace\test3\src\plugins\e57\libE57Format\src\ImageFileImpl.cpp) [D:\dev\workspace\test3\build\plugins\e57\libE57Format\E57Format.vcxproj]

--- a/plugins/e57/libE57Format/CMakeLists.txt
+++ b/plugins/e57/libE57Format/CMakeLists.txt
@@ -79,7 +79,7 @@ target_compile_options(E57Format PRIVATE
     )
 endif()
 target_link_libraries(E57Format
-    PUBLIC
+    PRIVATE
         ${XercesC_LIBRARIES}
 )
 
@@ -88,6 +88,7 @@ target_include_directories(E57Format
     PRIVATE
         include
         contrib/CRCpp/inc
+        ${XercesC_INCLUDE_DIR}
 )
 
 target_compile_definitions(E57Format

